### PR TITLE
 feat(Coupon): add backend call to coupon apply btn 

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- The apply button should validate the coupon entered by the user

## This pull request

- Calls the invoice preview endpoint with the priceId and coupon code
  when the Apply button is pressed.

## Issue that this pull request solves

Closes: #10850

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Comment by StaberindeZA** [comment]
For testing you can use the following promotion code and plan.
- Code: ASJQLIUQ
- Plan: price_1JfcqJBVqmGyQTMadYoEH25q.

File: packages/fxa-payments-server/src/lib/apiClient.test.ts
Comment (chenba) [inline packages/fxa-payments-server/src/lib/apiClient.test.ts:499]: This test seems unnecessary.  It's the same test as the one above.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx
Comment (chenba) [inline packages/fxa-payments-server/src/components/CouponForm/index.tsx:59]: If I recall from the other PR, setting error to true here would render the error message about the promotion code being invalid or expired.  But the `err` caught here could be an API error, and in that case, we'd be misinforming the user.

**Comment by biancadanforth** [comment]
@StaberindeZA , can you rebase this off of `main` since Ivo's PR merged? It'd make it easier to review.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx
Comment (biancadanforth) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx:59]: I just tested this, and Barry is correct. It's kind of the difference between a 400 and a 500 server response.

I don't think it needs to block this PR, but perhaps it's worth discussing in the next design check-in if we want to add at least one other generic error message here in the event there was a server error (i.e. nothing is wrong with the code the user entered, and nothing they can do, perhaps other than wait for us to fix it, will change the result). Can you add a topic to the agenda to this effect Reino?

At the very least for this PR, perhaps we can capture the error in Sentry if it's an API error?

**Review by biancadanforth — APPROVED** [review]
R + WC (with changes). Thanks for the quick PR here Reino. I think it's looking pretty good (and it works!). There are a number of minor issues that I request inline, and a few overall:
* Please rebase off the latest `main` and resolve the merge conflicts.
* Looks like a coupon-related test is still failing in CI.
* The issue referenced in the commit description is not the correct issue (should be #10850 ); looks like the GitHub/Jira sync broke somehow.
* Nit: Can you rename the commit message to "feat(payments): …", since it’s on the payments server? This commit message is parsed in `release.sh` for generating our changelogs.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx
Comment (StaberindeZA) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx:26]: Yup that makes sense. I'll make the change.

File: packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx
Comment (StaberindeZA) [reply packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx:10]: Ah sweet, didn't know we had this. Thank you!

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx
Comment (StaberindeZA) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: I thought it best to send the error to Sentry in the `checkCouponCode` (`checkPromotionCode`) function.

I did update the logic a bit to send a different error message. I'll create a separate issue to properly implement this, but at least we'll have something for now.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx
Comment (StaberindeZA) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx]: I actually think the linter prefers without the brackets.

File: packages/fxa-payments-server/src/components/PlanDetails/index.tsx
Comment (IvoJP) [inline packages/fxa-payments-server/src/components/PlanDetails/index.tsx:55]: small nit in case you get to bianca's before this is merged, can we remove the console.log statement to keep the console clean? otherwise, non-blocker.

File: packages/fxa-payments-server/server/config/index.js
Comment (chenba) [inline packages/fxa-payments-server/server/config/index.js:5]: Was this line breaking something?

File: packages/fxa-payments-server/src/components/CouponForm/index.test.tsx
Comment (chenba) [inline packages/fxa-payments-server/src/components/CouponForm/index.test.tsx:33]: Another way is to wrap the code that causing React rendering in `act()`.  E.g. `await act(async () => fireEvent.click(someButton));`.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx
Comment (chenba) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx:27]: > Style on this `if` block is also off (missing brackets).
> 
> Also, unlike the auth server, the payments server reports all errors to Sentry, so I don't think you need to differentiate here if it's only for the purposes of capturing the error in Sentry.

The calling code catches this error, so I think explicit Sentry reporting would be needed.

Comment (StaberindeZA) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx:27]: OK cool. I've left it in for now.

File: packages/fxa-payments-server/server/config/index.js
Comment (StaberindeZA) [reply packages/fxa-payments-server/server/config/index.js:5]: Not that I can think of. I think I touched this file while debugging, and the linter auto deleted this. Should I put it back?

File: packages/fxa-payments-server/src/components/CouponForm/index.test.tsx
Comment (StaberindeZA) [reply packages/fxa-payments-server/src/components/CouponForm/index.test.tsx:33]: So I tried doing that, but for some reason it wasn't working. It was late at night though, so I might have done something else wrong. 😅

File: packages/fxa-payments-server/server/config/index.js
Comment (chenba) [reply packages/fxa-payments-server/server/config/index.js:5]: Nah.  Just curious on the change.

File: packages/fxa-payments-server/src/components/CouponForm/index.test.tsx
Comment (chenba) [reply packages/fxa-payments-server/src/components/CouponForm/index.test.tsx:33]: Works > doesn't work. 😄

**Pull request merged**
